From 233808a58db1f62d773b03f9dad599924170aca6 Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Wed, 18 Mar 2015 08:36:37 +0000
Subject: lib/batch: Trivial compile fix for 32-bit builds

intel_batchbuffer.c: In function 'fill_object':
intel_batchbuffer.c:589:20: error: cast from pointer to integer of different size [-Werror=pointer-to-int-cast]
  obj->relocs_ptr = (uint64_t)relocs;
                    ^
intel_batchbuffer.c: In function 'exec_blit':
intel_batchbuffer.c:598:21: error: cast from pointer to integer of different size [-Werror=pointer-to-int-cast]
  exec.buffers_ptr = (uint64_t)objs;

Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>

diff --git a/lib/intel_batchbuffer.c b/lib/intel_batchbuffer.c
index 666c323..c1c27a6 100644
--- a/lib/intel_batchbuffer.c
+++ b/lib/intel_batchbuffer.c
@@ -548,7 +548,7 @@ fill_object(struct drm_i915_gem_exec_object2 *obj, uint32_t gem_handle,
	memset(obj, 0, sizeof(*obj));
	obj->handle = gem_handle;
	obj->relocation_count = count;
-	obj->relocs_ptr = (uint64_t)relocs;
+	obj->relocs_ptr = (uintptr_t)relocs;
 }

 static void exec_blit(int fd,
@@ -557,7 +557,7 @@ static void exec_blit(int fd,
 {
	struct drm_i915_gem_execbuffer2 exec;

-	exec.buffers_ptr = (uint64_t)objs;
+	exec.buffers_ptr = (uintptr_t)objs;
	exec.buffer_count = count;
	exec.batch_start_offset = 0;
	exec.batch_len = batch_len * 4;
